### ISSUE TITLE
CodeGen missing types when an `interface` contains a `union` type.

### ISSUE THREAD

**Issue by morluna** [apollographql/apollo-ios#2198]
## Bug report

Given the following schema:
```graphqls
type OverviewCardTitle {
  """The icon on the title."""
  icon: AppImage

  """The title text."""
  text: String!

  """The subtitle for the title."""
  subtitle: String
}

type OverviewInvestCardContentSubtitle {
  """The change for all of the user's invest accounts."""
  change: Float

  """The percent change for all of the user's invest accounts."""
  percentChange: Float

  """Boolean declaring if this is a gain or loss."""
  isGain: Boolean
}

union OverviewContentBaseSubtitle = OverviewCardTitle | OverviewInvestCardContentSubtitle

interface OverviewCardContentIterface {
  """The text for the content."""
  text: String!

  """The subtitle for the content sent to the client."""
  subtitle: OverviewContentBaseSubtitle
}
```
When running `ApolloCodegenOptions`, we are unable to get `selection` for the `Subtitle` node as shown below:

![Screen Shot 2022-03-14 at 4 33 21 PM](https://user-images.githubusercontent.com/8826948/158271871-2261ecf6-f5ba-40d2-979e-ffbd8c31da68.png)

However if we change the `OverviewCardContentIterface` to a `type`, we do get the `Subtitle.selections` generated code:
```graphqls
type OverviewCardContentIterface {
  """The text for the content."""
  text: String!

  """The subtitle for the content sent to the client."""
  subtitle: OverviewContentBaseSubtitle
}
```

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 0.51.1
- Xcode version: 13.2.1
- Swift version: 5.5.2
- Package manager: ??

**Comment by calvincestari** [comment]
Thanks for reporting this @morluna, we'll take a look into it and get back to you.

**Comment by calvincestari** [comment]
Hi 👋🏻 - this issue should be resolved in 1.0. 1.0 is currently in alpha releases with a full 1.0 release in H2 2022. Please take a look at the [latest alpha](https://github.com/apollographql/apollo-ios/releases) and let us know if this is still an issue for you.
